packages/SwingSet/src/liveslots/collectionManager.js

```diff
@@ -509,10 +515,62 @@ export function makeCollectionManager(
       return iter();
     }
 
+    /**
+     * Clear the entire contents of a collection non-selectively.  Since we are
+     * being unconditional, we don't need to inspect any of the keys to decide
+     * what to do and therefor can avoid deserializing the keys.  In particular,
+     * this avoids swapping in any virtual objects that were used as keys, which
+     * can needlessly thrash the virtual object cache when an entire collection
+     * is being deleted.
+     *
+     * @returns {boolean} true if this operation introduces a potential
+     *   opportunity to do further GC.
+     */
+    function clearInternalFull() {
+      let doMoreGC = false;
+      const [coverStart, coverEnd] = getRankCover(M.any(), encodeKey);
+      let priorDBKey = '';
+      const start = prefix(coverStart);
+      const end = prefix(coverEnd);
+      while (priorDBKey !== undefined) {
+        const [dbKey, dbValue] = syscall.vatstoreGetAfter(
+          priorDBKey,
+          start,
+          end,
+        );
+        if (!dbKey) {
+          break;
+        }
+        if (dbKey < end) {
+          priorDBKey = dbKey;
+          const value = JSON.parse(dbValue);
+          doMoreGC =
+            value.slots.map(vrm.removeReachableVref).some(b => b) || doMoreGC;
+          syscall.vatstoreDelete(dbKey);
+          if (dbKey[0] === 'r') {
+            const keyVref = vrefFromDBKey(dbKey);
+            if (hasWeakKeys) {
+              vrm.removeRecognizableVref(keyVref, `${collectionID}`, true);
+            } else {
+              doMoreGC = vrm.removeReachableVref(keyVref) || doMoreGC;
+            }
+            syscall.vatstoreDelete(prefix(`|${keyVref}`));
+          }
+        }
+      }
+      if (!hasWeakKeys) {
+        syscall.vatstoreSet(prefix('|entryCount'), '0');
```

Comment:
This adds a vatstoreSet to the "delete the whole collection" case that wasn't there before, right? That seems like a tiny regression.

What do you think about having the caller (`clearInternal`) retain responsibility for writing out this length=0, so it can continue to use the same `if (!hasWeakKeys && !isDeleting)` logic as before? In that case, `clearInternal()` would do something like:
```js
{
  let doMoreGC = false;
  if (isDeleting || (matchAny(keyPatt) && matchAny(valuePatt))) {
    doMoreGC = clearInternalFull();
  } else {
    for (const k of keys(keyPatt, valuePatt)) {
      doMoreGC = deleteInternal(k) || doMoreGC;
    }
  }
  if (!hasWeakKeys && !isDeleting) {
...
```

Reply:
That makes sense.